components/tidb_query_expr/src/impl_like.rs

```diff
@@ -66,6 +70,87 @@ pub fn like<C: Collator>(target: BytesRef, pattern: BytesRef, escape: &i64) -> R
     Ok(Some(true as i64))
 }
 
+#[rpn_fn(capture = [metadata], metadata_mapper = init_regexp_utf8_data)]
+#[inline]
+pub fn regexp_utf8(
+    metadata: &Option<Regex>,
+    target: BytesRef,
+    pattern: BytesRef,
+) -> Result<Option<i64>> {
+    let target = match String::from_utf8(target.to_vec()) {
+        Ok(target) => target,
+        Err(err) => return Err(box_err!("invalid input value: {:?}", err)),
+    };
+
+    match metadata {
+        Some(regex) => Ok(Some(regex.is_match(target.as_ref()) as i64)),
+        None => {
+            let pattern = match String::from_utf8(pattern.to_vec()) {
+                Ok(pattern) => pattern,
+                Err(err) => return Err(box_err!("invalid input value: {:?}", err)),
+            };
+            let pattern = format!("(?i){}", pattern);
+
+            match BytesRegex::new(&pattern) {
```

Comment:
Why we need to use `BytesRegex` for utf-8 here? Both `target` and `pattern` should be valid utf-8 value here.